scipy/_lib/tests/test_warnings.py

```diff
@@ -40,7 +40,21 @@ def visit_Call(self, node):
         ast.NodeVisitor.generic_visit(self, node)
 
         if p.ls[-1] == 'simplefilter' or p.ls[-1] == 'filterwarnings':
-            if node.args[0].value == "ignore":
+            # get first argument of the `args` node of the filter call
+            match node.args[0]:
+                case ast.Constant() as c:
+                    argtext = c.value
+                case ast.JoinedStr() as js:
+                    # if we get an f-string, discard the templated pieces, which
+                    # are likely the type or specific message; we're interested
+                    # in the action, which is less likely to use a template
+                    argtext = "".join(
+                        x.value for x in js.values if isinstance(x, ast.Constant)
+                    )
+                case _:
+                    raise ValueError("unknown ast node type")
+            # check if filter is set to ignore
+            if argtext.split(":")[0] == "ignore":
```

Comment:
Guess I shouldn't have taught the parser here to be smarter about finding ignored warnings. It finds quite a few of them now 😅 

Probably worth splitting this off into a separate PR...